pkg/detectors/postgres/postgres_test.go

```diff
@@ -0,0 +1,336 @@
+				data: []byte(fmt.Sprintf(`
+					POSTGRES_USER=%s
+					POSTGRES_PASSWORD=%s
+					POSTGRES_ADDRESS=%s
+					POSTGRES_PORT=%s
+					`, postgresUser, inactivePass, postgresHost, postgresPort)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found with seperated credentials - no port, unverified",
+			s:    Scanner{},
+			args: args{
+				ctx: context.Background(),
+				data: []byte(fmt.Sprintf(`
+					POSTGRES_USER=%s
+					POSTGRES_PASSWORD=%s
+					POSTGRES_ADDRESS=%s
+					`, postgresUser, inactivePass, postgresHost)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found with single line credentials, unverified",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf(`postgresql://%s:%s@%s:%s/postgres`, postgresUser, inactivePass, postgresHost, postgresPort)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found with json credentials, unverified - inactive password",
+			s:    Scanner{},
+			args: args{
+				ctx: context.Background(),
+				data: []byte(fmt.Sprintf(
+					`DB_CONFIG={"user": "%s", "password": "%s", "host": "%s", "port": "%s", "database": "postgres"}`, postgresUser, inactivePass, postgresHost, postgresPort)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found with json credentials, unverified - inactive user",
+			s:    Scanner{},
+			args: args{
+				ctx: context.Background(),
+				data: []byte(fmt.Sprintf(
+					`DB_CONFIG={"user": "%s", "password": "%s", "host": "%s", "port": "%s", "database": "postgres"}`, inactiveUser, postgresPass, postgresHost, postgresPort)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found, unverified due to error - inactive port",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf(`postgresql://%s:%s@%s:%s/postgres`, postgresUser, postgresPass, postgresHost, inactivePort)),
+				verify: true,
+			},
+			want: func() []detectors.Result {
+				r := detectors.Result{
+					DetectorType: detectorspb.DetectorType_Postgres,
+					Verified:     false,
+				}
+				return []detectors.Result{r}
+			}(),
+			wantErr: false,
+		},
+		// This test seems take a long time to run (70s+) even with the timeout set to 1s. It's not clear why.
```

Comment:
inactive/incorrect hosts always seem to take a long time even with timeout (70s+). not sure why. this could present a problem when running. any ideas why this might happen or how to mitigate?

Reply:
which call specifically is timing out?

Reply:
[this](https://github.com/trufflesecurity/trufflehog/pull/2108/files#diff-655d62de1dd6db263b159e4695761b8052314875331af5a1e97e5c95e2216f64R218) one

Reply:
yep, good call. added.